File: platforms/web/examples/scene/material/normal_texture.js
import { Material, Renderer } from "fragmentcolor";

const renderer = new Renderer();
const normal_map_bytes = [ 128, 128, 255, 255, 128,   128, 255, 255, 128,   128, 255, 255, 128,   128, 255, 255, ];
const normal_map = await renderer.createTexture(normal_map_bytes, [2, 2]);
const mat = Material.pbr().normalTexture(normal_map).normalScale(1.2);

File: platforms/kotlin/examples/scene/material/normal_texture.kt
import org.fragmentcolor.*

val renderer = Renderer()
val normal_map_bytes = listOf(128.0f, 128.0f, 255.0f, 255.0f, 128.0f, 128.0f, 255.0f, 255.0f, 128.0f, 128.0f, 255.0f, 255.0f, 128.0f, 128.0f, 255.0f, 255.0f, .0f)
val normal_map = renderer.createTexture(TextureInputMobile.Bytes(normal_map_bytes.let { ba -> ByteArray(ba.size) { i -> ba[i].toInt().and(0xFF).toByte() } }), null)
val mat = Material.pbr().normalTexture(normal_map).normalScale(1.2f)